commons/zenoh-shm/src/api/protocol_implementations/posix/posix_shm_provider_backend.rs

```diff
@@ -12,289 +12,10 @@
 //   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
 //
 
-use std::{
-    cmp,
-    collections::BinaryHeap,
-    num::NonZeroUsize,
-    sync::{
-        atomic::{AtomicUsize, Ordering},
-        Arc, Mutex,
-    },
+use crate::api::protocol_implementations::posix::posix_shm_provider_backend_talc::{
+    PosixShmProviderBackendTalc, PosixShmProviderBackendTalcBuilder,
 };
 
-use zenoh_core::{zlock, Resolvable, Wait};
-use zenoh_result::ZResult;
+pub type PosixShmProviderBackendBuilder<What> = PosixShmProviderBackendTalcBuilder<What>;
```

Comment:
If this is designed for switching, shall we define it as an enum that is Talc by default?